tests/test_information_from_events.py

```diff
@@ -0,0 +1,464 @@
+        currency_network_contract, transfer_event
+    ):
+        path_from_events.append(event["args"]["_to"])
+
+    return path_from_events
+
+
+def get_previous_balance(balance_updates, balance_update_event):
+    """Returns the balance before a given balance update event"""
+    index = 0
+    # find the corresponding event
+    for i, (key, value) in enumerate(balance_updates):
+        if key == (
+            balance_update_event["blockNumber"],
+            balance_update_event["logIndex"],
+            balance_update_event["transactionHash"],
+        ):
+            index = i
+            break
+    else:
+        raise RuntimeError("Could not find balance update")
+    index -= 1
+
+    if index < 0:
+        return 0
+
+    return balance_updates[index][1]
+
+
+def get_interest_at(currency_network_contract, balance_update_event):
+    """Returns the occurred interests at a given balance event"""
+    from_ = balance_update_event["args"]["_from"]
+    to = balance_update_event["args"]["_to"]
+    interests = get_interests_for_trustline(currency_network_contract, from_, to)
+    balance_updates = get_balance_updates_for_trustline(
+        currency_network_contract, from_, to
+    )
+    index = 0
+    for i, (key, value) in enumerate(balance_updates):
+        if key == (
+            balance_update_event["blockNumber"],
+            balance_update_event["logIndex"],
+            balance_update_event["transactionHash"],
+        ):
+            index = i
+            break
+    else:
+        raise RuntimeError("Could not find balance update")
+
+    index -= 1
+    if index < 0:
+        return 0
+
+    return interests[index]
+
+
+def get_delta_balances_of_transfer(currency_network_contract, transfer_event):
+    """Returns the balance changes along the path because of a given transfer"""
+    balance_update_events = get_transfer_balance_update_events(
+        currency_network_contract, transfer_event
+    )
+
+    post_balances = []
+    for event in balance_update_events:
+        post_balances.append(event["args"]["_value"])
+
+    pre_balances = []
+    for event in balance_update_events:
+        from_ = event["args"]["_from"]
+        to = event["args"]["_to"]
+        pre_balance = get_previous_balance(
+            get_balance_updates_for_trustline(currency_network_contract, from_, to),
+            event,
+        )
+        pre_balances.append(pre_balance)
+
+    interests = []
+    for event in balance_update_events:
+        interest = get_interest_at(currency_network_contract, event)
+        interests.append(interest)
+
+    # sender balance change
+    delta_balances = [post_balances[0] - pre_balances[0] - interests[0]]
+
+    # mediator balance changes
+    for i in range(len(balance_update_events) - 1):
+        delta_balances.append(
+            (post_balances[i + 1] - pre_balances[i + 1] - interests[i + 1])
+            - (post_balances[i] - pre_balances[i] - interests[i])
```

Comment:
This is a pain to understand for me, because for a user `i+1` in the path, `post_balances[i+1]` corresponds to the balance of the next TL in the path based on my point of view, whereas `post_balances[i]` corresponds to the balance of the previous TL in the path based on the previous user's point of view.

I would do:
```
    # mediator balance changes
    balance_change_previous_tl = delta_balances[0]
    for i in range(1, len(balance_update_events)):
        balance_change_next_tl = post_balances[i] - pre_balances[i] - interests[i]
        delta_balances.append(balance_change_next_tl - balance_change_previous_tl)
        balance_change_previous_tl = balance_change_next_tl

    # receiver balance change
    delta_balances.append(-balance_change_previous_tl)

```

Reply:
yes, exactly like it is for your code (balance_change_next_tl is from your point of view, balance_change_previous_tl is from the view of the guy before you) so I do not see your point?
I think your code is harder to understand, because it connects two steps of the loop where es in my code you only every have to think about the current step. Also I would be not sure if previous means previous iteration or previous hop. I added your variable names, so maybe it is better two understand now, that I am actually doing the same thing.